### ISSUE TITLE
SyntaxError: not a TIFF IFD - Is this expected?

### ISSUE THREAD

**Issue by jplehmann** [python-pillow/Pillow#2043]
### What did you do?

Using sorl.thumbnail to generate a thumbnail, which uses Pillow backend for obtaining the EXIF info to determine orientation. 

### What did you expect to happen?

To get EXIF info with no errors.

### What actually happened?

```
  File "/Users/john/.venv/mm/lib/python2.7/site-packages/sorl/thumbnail/base.py", line 157, in _create_thumbnail
    image = default.engine.create(source_image, geometry, options)
  File "/Users/john/.venv/mm/lib/python2.7/site-packages/sorl/thumbnail/engines/base.py", line 20, in create
    image = self.orientation(image, geometry, options)
  File "/Users/john/.venv/mm/lib/python2.7/site-packages/sorl/thumbnail/engines/base.py", line 45, in orientation
    return self._orientation(image)
  File "/Users/john/.venv/mm/lib/python2.7/site-packages/sorl/thumbnail/engines/pil_engine.py", line 71, in _orientation
    exif = image._getexif()
  File "/Users/john/.venv/mm/lib/python2.7/site-packages/PIL/JpegImagePlugin.py", line 390, in _getexif
    return _getexif(self)
  File "/Users/john/.venv/mm/lib/python2.7/site-packages/PIL/JpegImagePlugin.py", line 418, in _getexif
    info = TiffImagePlugin.ImageFileDirectory(head)
  File "/Users/john/.venv/mm/lib/python2.7/site-packages/PIL/TiffImagePlugin.py", line 274, in __init__
    raise SyntaxError("not a TIFF IFD")
SyntaxError: not a TIFF IFD
```
### What versions of Pillow and Python are you using?

Pillow==2.6.1
Python 2.7.10

Please include code that reproduces the issue and whenever possible, an image that demonstrates the issue. The best reproductions are self-contained scripts with minimal dependencies.

```
i = Image.open("...")
i._getexif()
```

I'm attaching the image causing the problem.  

![20160707034802-20160628_174527_wm](https://cloud.githubusercontent.com/assets/460964/17211874/6767818c-5491-11e6-9c8d-584c2f608dd0.jpg)

Note that this image has a watermark applied.  I had no problems with the image BEFORE the watermark was applied so clearly that's doing something bad/unsupported.

When I use `mdls` to view the EXIF info all looks well to me.  It dumps lots of metadata including orientation to the console.

Bottom line, I don't know if Pillow should support this image's EXIF info or if the image is sort-of corrupt.  Thanks!

**Comment by wiredfool** [comment]
The exif looks corrupt to me. 

```
im = Image.open('6767818c-5491-11e6-9c8d-584c2f608dd0.jpg')
>>> im._getexif()
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "build/bdist.linux-x86_64/egg/PIL/JpegImagePlugin.py", line 393, in _getexif
  File "build/bdist.linux-x86_64/egg/PIL/JpegImagePlugin.py", line 426, in _getexif
  File "build/bdist.linux-x86_64/egg/PIL/TiffImagePlugin.py", line 815, in __init__
  File "build/bdist.linux-x86_64/egg/PIL/TiffImagePlugin.py", line 443, in __init__
SyntaxError: not a TIFF file (header '\x00\x00*\x00\x08\x00\x00\x00' not valid)
>>> im.info
{'jfif_version': (1, 1), 'jfif': 257, 'jfif_unit': 0, 'jfif_density': (1, 1), 'exif': 'Exif\x00\x00\x00\x00*\x00\x08\x00\x00\x00\x01\x00i\x87\x04\x00\x01\x00\x00\x00\x1c\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00'}
```

For reference, a valid exif looks like:
```
 im.info['exif']
'Exif\x00\x00MM\x00*\x00\x00\x00\x08\x00\x0c\x01\x00\x00\x03\x00\x00\x00\x01...
```

The exif you have is mssling the byte order marks, MM or II

**Comment by jplehmann** [comment]
Thank you for your speedy response. I will find out what is generating the watermark and report the bug.  I'll report back if they have some kind of argument.

**Comment by yeyuqiu** [comment]
File "/opt/app/python/lib/python2.7/site-packages/sorl/thumbnail/shortcuts.py", line 8, in get_thumbnail
    return default.backend.get_thumbnail(file_, geometry_string, **options)
  File "/opt/app/python/lib/python2.7/site-packages/sorl/thumbnail/base.py", line 125, in get_thumbnail
    thumbnail)
  File "/opt/app/python/lib/python2.7/site-packages/sorl/thumbnail/base.py", line 157, in _create_thumbnail
    image = default.engine.create(source_image, geometry, options)
  File "/opt/app/python/lib/python2.7/site-packages/sorl/thumbnail/engines/base.py", line 20, in create
    image = self.orientation(image, geometry, options)
  File "/opt/app/python/lib/python2.7/site-packages/sorl/thumbnail/engines/base.py", line 45, in orientation
    return self._orientation(image)
  File "/opt/app/python/lib/python2.7/site-packages/sorl/thumbnail/engines/pil_engine.py", line 71, in _orientation
    exif = image._getexif()
  File "/opt/app/python/lib/python2.7/site-packages/PIL/JpegImagePlugin.py", line 395, in _getexif
    return _getexif(self)
  File "/opt/app/python/lib/python2.7/site-packages/PIL/JpegImagePlugin.py", line 428, in _getexif
    info = TiffImagePlugin.ImageFileDirectory_v1(head)
  File "/opt/app/python/lib/python2.7/site-packages/PIL/TiffImagePlugin.py", line 818, in __init__
    ImageFileDirectory_v2.__init__(self, *args, **kwargs)
  File "/opt/app/python/lib/python2.7/site-packages/PIL/TiffImagePlugin.py", line 443, in __init__
    raise SyntaxError("not a TIFF file (header %r not valid)" % ifh)
SyntaxError: not a TIFF file (header '' not valid)

Versions of Pillow and Python:
sorl-thumbnail (12.3)
Python 2.7.6
Pillow (3.4.2)

Here's the problem image: http://www.eyaos.com/static/upload/skupic/2016/05/29/c6dad4bf-e881-4797-ab44-f75d1f2753f6.jpg
